packages/client/lib/sync/fetcher/blockfetcher.ts

```diff
@@ -22,9 +22,12 @@ export class BlockFetcher extends BlockFetcherBase<Block[], Block> {
    * @param job
    */
   async request(job: Job<JobTask, Block[], Block>): Promise<Block[]> {
-    const { task, peer } = job
-    const { first, count } = task
-
+    const { task, peer, partialResult } = job
+    let { first, count } = task
+    if (partialResult) {
```

Comment:
i think we should add some test coverage here, since it's a delicate part, and codecov is reporting missing

Reply:
thanks for adding!!